src/panels/config/automation/blueprint-automation-editor.ts

```diff
@@ -195,20 +186,9 @@ export class HaBlueprintAutomationEditor extends LitElement {
     }
     const input = { ...this.config.use_blueprint.input, [key]: value };
 
-    const blueprint = this._blueprint;
-    const metaValue =
-      !blueprint || "error" in blueprint
-        ? undefined
-        : blueprint?.metadata.input && blueprint?.metadata?.input[key];
-    const keyDefault = metaValue && metaValue.default;
-
-    if ((value === "" && !keyDefault) || value === undefined) {
-      delete input[key];
```

Comment:
I'm not sure why we ever needed to delete input[key] when it was undefined. Undefined keys are already filtered out of the yaml, so leaving them in the config seems harmless, and that helps us know when to populate the field with default or not.